src/components/TextInput/BaseTextInput.js

```diff
@@ -167,17 +167,26 @@ function BaseTextInput(props) {
         [props.autoGrowHeight, props.multiline],
     );
 
+    const hasValueRef = useRef(inputValue.length > 0);
```

Comment:
@parasharrajat I added a few comments to the code and moved it a bit to make it easier to understand. I hope it's better now?

In general we can't use inputValue directly, because it might doesn't represent the actual input value, due to the following reasons:

- `props.value` might be undefined, when its a uncontrolled text component
- `inputValue` is composed out of `props.defaultValue`. The input might get cleared of the `defaultValue`, but `inputValue` would still contain it as it doesn't get updated.

Thus we have to keep track of the actual fact whether there is a value inside the input field or not, which is what this ref accomplishes.
It's a ref because we want to avoid unnecessary re-renders.